### ISSUE TITLE
HvdAllToAllEmbedding 多cpu分布训练时，梯度貌似没有聚合

### ISSUE THREAD

**Issue by Rebecca-rr** [tensorflow/recommenders-addons#499]
使用的环境
TFRA 0.7.2 
TensorFlow 2.15.1
horovod 0.28.1 

HvdAllToAllEmbedding 的描述中
”In the backward training, the gradient of sparse parameters will be distributed to other cards in the default modulo operator way through AllToAll communication between cards. After each card obtains its own sparse gradient, the sparse optimizer will be executed to complete the optimization of large-scale sparse features.“

但是源码中
只看到了
1）HvdAllToAllEmbedding 前向 embedding_lookup中使用了self.hvd.alltoall的聚合。 
2）dynamic_embedding_optimizer.py 中只对模型变量的梯度做了聚合。
并没有看到HvdAllToAllEmbedding 对应的梯度在反向传播中聚合。

所以很困惑，多cpu  分布式训练下，backward training， embedding的gradient是否会聚合，以什么样的方式聚合？同步还是异步 ？sum or mean ?

辛苦解惑~~~
